futures-macro/src/stream_select.rs

```diff
@@ -0,0 +1,110 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse::Parser, punctuated::Punctuated, Expr, Index, Token};
+
+/// The `stream_select!` macro.
+pub(crate) fn stream_select(input: TokenStream) -> TokenStream {
+    let args = Punctuated::<Expr, Token![,]>::parse_terminated
+        .parse(input)
+        .expect("macro expects a comma separated list of expressions");
```

Comment:
I would like to avoid panic on proc-macro. Could you change stream_select function to return Result and handle it on the top-level stream_select_internal function? Like:

```rs
pub fn stream_select_internal(input: TokenStream) -> TokenStream {
    crate::stream_select::stream_select(input).unwrap_or_else(Error::into_compile_error)
}
```